### ISSUE TITLE
Updating self.main_window.content multiple times raises a RuntimeError

### ISSUE THREAD

**Issue by martinbarnard** [beeware/toga#1509]
**Describe the bug**
When trying to change `main_window.content` twice a RuntimeError occurs with the message: "Already have a container"
**Note**: Error is the same with separate handlers for button press events. Sample code re-uses same button press handler for simplicity.

**To Reproduce**
 * Create a sample "HelloWorld" application with briefcase
 * Edit app.py:
```python
"""
My first application
"""
import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW


class HelloWorld(toga.App):

    window_list = {}
    def startup(self):
        main_box = toga.Box()
        second_box = toga.Box()
        def button_handler(widget):
            data = self.window_list[widget.label]
            self.main_window.content = data
            self.main_window.show()

        btn = toga.Button('second', on_press = button_handler)
        btn2 = toga.Button('main', on_press = button_handler)
        self.window_list['main'] = main_box
        self.window_list['second'] = second_box
        main_box.add(btn)
        second_box.add(btn2)


        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()


def main():
    return HelloWorld()


```
**Expected behavior**
Button to toggle between "maiin" and "second" upon each press of the button

**Screenshots**
n/a

**Environment:**
 - Linux Mint 20
 - 3.8
 - Software versions:
   - Briefcase: 0.3.7
   - Toga: 0.3.0.dev34

   - ...

**Additional context**

Full error message:
```bash
/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga_gtk/app.py:40: DeprecationWarning: Gtk.Window.set_wmclass is deprecated
  self.native.set_wmclass(app.interface.name, app.interface.name)
Traceback (most recent call last):
  File "/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga_gtk/widgets/button.py", line 37, in gtk_on_press
    self.interface.on_press(self.interface)
  File "/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga/handlers.py", line 66, in _handler
    result = handler(interface, *args, **kwargs)
  File "/home/martinb/src/beeware/test_example/helloworld/src/helloworld/app.py", line 18, in button_handler
    self.main_window.content = data
  File "/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga/window.py", line 137, in content
    self._impl.set_content(widget._impl)
  File "/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga_gtk/window.py", line 116, in set_content
    child._impl.container = widget
  File "/mnt/sdb/cache/.cache/pypoetry/virtualenvs/beeware-rTajD2UR-py3.8/lib/python3.8/site-packages/toga_gtk/widgets/base.py", line 35, in container
    raise RuntimeError('Already have a container')
RuntimeError: Already have a container

```

**Comment by mhsmith** [comment]
This happens when assigning the same widget to `window.content` multiple times. So this is a more specific issue than #1439, which is about whether assigning to `window.content` works at all.

It can be worked around by setting `window.content` to a permanent top-level container box, and then adding and removing things through that:

```python
import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW


class HelloWorld(toga.App):

    window_list = {}
    def startup(self):
        container_box = toga.Box()
        main_box = toga.Box()
        second_box = toga.Box()
        def button_handler(widget):
            while container_box.children:
                container_box.remove(container_box.children[-1])
            data = self.window_list[widget.label]
            container_box.add(data)

        btn = toga.Button('second', on_press = button_handler)
        btn2 = toga.Button('main', on_press = button_handler)
        self.window_list['main'] = main_box
        self.window_list['second'] = second_box
        main_box.add(btn)
        second_box.add(btn2)
        container_box.add(main_box)

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = container_box
        self.main_window.show()


def main():
    return HelloWorld()
```

**Comment by mhsmith** [comment]
It looks like this affects every platform, because they all have very similar code in widgets/base.py:

https://github.com/beeware/toga/blob/100e624437b02a3e6eec34b0b71f8e5342e506eb/src/gtk/toga_gtk/widgets/base.py#L31-L37

I don't understand why it would be an error to replace the container, when there's nothing stopping you from doing the same thing in two separate steps: first removing it from one container and then adding it to another one. I'll create a PR to explore fixing this.

**Comment by saroad2** [comment]
This bug was fixed in #1572 .

If you see it returns, please feel free to reopen the issue!
